### PR TITLE
Jvenstad/freeze to block window

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java [hunk 1]
```diff
@@ -313,28 +313,37 @@ private ApplicationController applications() {
...
                             completedAt = stepJobs.stream().map(job -> application.deploymentJobs().statusOf(job).get().lastCompleted().get().at()).max(naturalOrder());
                             reason = "Available change in " + stepJobs.stream().map(JobType::jobName).collect(joining(", "));
                         }
                     }
                 }
             }
-            if (testJobs == null) {
-                testJobs = testJobs(application, Versions.from(application, controller.systemVersion()),
+            if (testJobs == null) { // If nothing to test, but outstanding commits, test those.
+                Change latestChange = application.outstandingChange().application()
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: You managed to find another ingenious Optional filter sequence that made me scratch my head. Well done!  :-)

How about
```
                Optional<ApplicationVersion> outstandingChange = application.outstandingChange().application();
                Change latestChange = outstandingChange.isPresent() ? application.change().with(outstandingChange.get()) 
                                                                    : application.change();
```

**Review by bratseth — APPROVED** [review]
Great! Check my comments and merge at will.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java [hunk 2]
```diff
@@ -77,14 +77,14 @@ public ApplicationList notUpgradingTo(Optional<Version> version) {
         return notUpgradingTo(version.get());
     }
 
-    /** Returns the subset of applications which are currently deploying a change */
+    /** Returns the subset of applications which have unfinished changes */
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationList.java]: Yes, this occurred to me as well, on the bike :)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java]: Hmm ... it felt so natural. I'll consider your suggestion.

**Pull request merged**